### ISSUE TITLE
Inject delay?

### ISSUE THREAD

**Issue by adam900710** [iovisor/bcc#2336]
To my understanding, injecting delay into kernel kprobe point is not supported yet.

And it's not something like kprobe, injecting delay, especially sleepable delay can easily screw up certain work (e.g. function called with spinlock hold).

Is it possible to create some macro like ALLOW_INJECT_SLEEPABLE_DELAY() just like ALLOW_INJECT_ERROR() in kernel for that?
With that, it should be super useful to debug race related bug.

Thanks.

**Comment by yonghong-song** [comment]
No, as far as I know, bpf community has not discussed this yet.  This probably needs to be similar to injection such that a developer buy-in is needed. Also not sure how this could be implemented as sleep/block is not allowed in bpf program context. So this has to be happen outside the preempt_disable region. So this is probably not a bpf issue.

**Comment by Anjali05** [comment]
Has this been implemented yet?
